src/Store/Reducers/reducersHelper.js

```diff
@@ -64,21 +64,29 @@ export const addOrRemoveItemFromObj = (targetObj, inputArr) => {
     return { ...targetObj, ...inputObj };
 };
 
-export const getNewSelectedItems = (selectedItems, currentItems) => {
-    let payload = [].concat(selectedItems).map(({ id, selected }) => ({ id, selected }));
-    return addOrRemoveItemFromObj(currentItems, payload);
-};
-
 export const selectRows = (state, action) => {
-    const selectedUpdated = getNewSelectedItems(action.payload, state.selectedRows);
-    let prevLoadedRows = [].concat(state.prevLoadedRows);
-    let selectedRowsRawData = prevLoadedRows.filter(({ id }) => selectedUpdated[id] === true);
+    let selectedRows = state.selectedRows.slice();
+    if (Array.isArray(action.payload)) {
+        // when the state and payload are the same length it means that it is a "select none" action
+        if (state.selectedRows.length && state.selectedRows.length === action.payload.length) {
+            selectedRows = [];
+        } else {
+            selectedRows = action.payload;
+        }
+    } else {
+        const selectedRow = selectedRows.find(selected => selected.id === action.payload.id);
+        if (selectedRow) {
+            selectedRows.splice(selectedRows.indexOf(selectedRow), 1);
```

Comment:
If you didn't use the solution from my comment above, this could be optimized, you see you're searching `selectedRows` twice, once with `find` and then second time with `indexOf` -- more optimal would be something like

```javascript
// findIndex() returns -1 if not found
const selectedRowIndex = selectedRows.findIndex(selected => selected.id === action.payload.id); 
if (selectedRowIndex !== -1) { // found
    selectedRows.splice(selectedRowIndex, 1);
}
else {
  ...
}
```